src/app/users/user-profile/user-profile.component.ts

```diff
@@ -18,11 +20,16 @@ export class UserProfileComponent implements OnInit {
   userId: string;
   questions: Question[];
   isMyProfile: boolean;
+  favoriteCourses$: Observable<Course[]>;
 
   constructor(public auth: AuthService, private db: DbService, private route: ActivatedRoute) {
     this.userId = this.route.snapshot.paramMap.get('uid');
     this.user$ = this.userId ? this.db.getUser(this.userId) : this.auth.user$;
     this.isMyProfile = (this.userId === null);
+    this.favoriteCourses$ = this.user$.pipe(
+      flatMap(user => this.db.getFavoriteCourses(user)),
+      map(courses => courses.sort((c1, c2) => (c1.name < c2.name) ? -1 : 1))
```

Comment:
Why name and not course number?

Reply:
because the name is shown in the list of buttons.

Reply:
Why don't we show the number too? Some courses are better recognized by their number

Reply:
It's an specification question. Let's open an issue and discuss, but it should not hold the PR.